packages/typescript-client/src/client.ts

```diff
@@ -319,10 +318,12 @@ export class ShapeStream<T extends Row = Row>
     return !this.isUpToDate
   }
 
-  #publish(messages: Message<T>[]) {
-    this.#subscribers.forEach(([subscriber, _]) => {
-      subscriber.process(messages)
-    })
+  async #publish(messages: Message<T>[]): Promise<void> {
+    await Promise.allSettled(
```

Comment:
We may want this to be `Promise.all`. If a callback throws we probably want to stop the stream rather than continue blindly pushing data into a callback that may be broken?

```suggestion
    await Promise.all(
```

Reply:
I'm trying to replicate the original behaviour which swallowed errors - the subscribe provides a callback whose errors the subscriber needs to handle themselves, the stream might have multiple subscribers

Reply:
Yeah pglite can stop the stream by aborting it directly in case there's a pglite error

Reply:
`allSettled` silently eats all errors, doesn't even show them in the console, and they don't bubble up to a global error handler such as `window.addEventListener('unhandledrejection',  () => {})`.

In PGlite we used `queueMicrotask` to lift the callback out of the current control flow. So something like:

```ts
  async #publish(messages: Message<T>[]): Promise<void> {
    await Promise.all(
      Array.from(this.#subscribers.values()).map(([callback, __]) =>
        queueMicrotask(() => callback(messages))
      )
    )
  }
```

Reply:
Although that now doesn't block while processing the callback

We can log the error, but again that doesn't bubble them to a global error handler:

```ts
  async #publish(messages: Message<T>[]): Promise<void> {
    await Promise.all(
      Array.from(this.#subscribers.values()).map(([callback, __]) =>
        try {
          callback(messages)
        } catch (error) {
          console.error(error)
        }
      )
    )
  }
```

Reply:
Final note, in the old electric we had code like this using `Promise.allSettled` for a callback and it silently eats errors in our CI causing a bug that was super hard to track down.

Reply:
With the `subscribe` call clients provide also an `onError` callback, but that's meant to provide actual stream errors, like network errors etc, not errors bubbled up from the callback itself, so I'm ruling out using that.

Other than that using `Promise.all` with a try catch block seems reasonable to me - errors should at least pop up in the console, I agree with that.

Reply:
So, would we support an optional `onError` callback to be passed to `subscribe`?
If it's provided we do `await callback().catch(onError)` and if no error callback is provided we re-throw or at least log the error.

Reply:
Subscribe already has an `onError` callback but it surfaces errors not related to the processing callback itself - if the developer wants to handle the callback errors they can wrap it in a try...catch statement and decide if they want to stop the subscription to the stream etc

I think we should just log the errors to console via `Promise.all` as I think that was the issue with `Promise.allSettled`

Reply:
I think the requirements should be that error in a callback is both logged to the console, but also makes it to any global error handler (which naturally gets logged to the console).

We don't need to handle that error, or pass it to an error handler. The user can internally wrap the callback with a try.

What we must not do is silently eat errors, and ideally let them bubble to the global handler.

Reply:
doing `Promise.all`with `try...catch` and bubbling error up with `queueMicrotask(() => { throw err })` - resolving this